lib/portlayer/exec/handle.go

```diff
@@ -211,9 +213,8 @@ func (h *Handle) Close() {
 	removeHandle(h.key)
 }
 
-func (h *Handle) SetState(s State) {
-	h.State = new(State)
-	*h.State = s
+func (h *Handle) SetContainerState(s State) {
```

Comment:
So this method is there to indicate a state change. This was not meant to change the cached state of the container, but to indicate during Commit if we want to change the state of the container.

We need two states here -- existing and new. This method should be to change the state. We can add an accessor for getting the current state -- that one could call h.Container.CurrentState().